MetaMask/metamask-extension
# fix: add new account after an account management snap is re-installed

## Explanation

Currently there's a bug when re-installing an account management snap and creating a new account. The new created account does not show up in MME.

This bug occurs because the `KeyringController` is purging the `SnapKeyring` if it has zero accounts, the next time a account management snap is installed and it tries to execute an action from the `SnapKeyring`, it will call the object referenced in the `metamask-controller` state and not the one inside the `KeyringController`. Therefore, the `KeyringController` has no way to know which accounts are part of `SnapKeyring` since it doesn't hold a reference to `SnapKeyring` inside it's own state.

* Fixes https://github.com/MetaMask/accounts-planning/issues/11

## Screenshots/Screencaps

https://github.com/MetaMask/metamask-extension/assets/17601467/57b1bb67-313a-40c6-92a9-77739cb6f865

### Before/After

There were no changes to the UI/UX.

## Manual Testing Steps

1. Install a keyring snap
2. Create an account with the snap
3. Notice that the list of accounts in the MME UI shows the new account
4. Uninstall the keyring snap
5. Notice that the account is gone from the MME UI
6. Re-install the same keyring snap
7. Create an account with the snap
8. The newly created account should appear in the MME UI

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
